util/buildkitd/buildkitd.go

```diff
@@ -237,6 +237,7 @@ func installBuildkit(ctx context.Context) error {
 		"--name", containerName,
 		"--privileged",
 		image+":"+vendoredVersion,
+		"--debug",
```

Comment:
My only remaining question is if it's worth forcing a restart of buildkitd when it's detected this flag is not there. I'm leaning on no and instead just letting it get reset when there's already going to be an upgrade or someone is starting fresh, but it should be possible to implement.

Reply:
I agree it's probably not worth it.

If we were to go down that path however, note that we kinda already do that for host networking:

https://github.com/dagger/dagger/blob/59a2c81c91a713749c255c2d465cd87042a9289c/util/buildkitd/buildkitd.go#L124

We retrieve information about the buildkit container and if it doesn't match what we want (buildkit version, or host networking), we nuke and restart. Shouldn't be too hard to include debug as well, but overkill IMHO.